### PR TITLE
Assert that unserialize makes only passables

### PR DESCRIPTION
**Pull request by erights**
Fixes https://github.com/Agoric/agoric-sdk/issues/4337.

Fixes https://github.com/Agoric/agoric-sdk/issues/4302 since thenables are not passable.

### CODE REVIEW

**Review by kriskowal — CHANGES_REQUESTED** [review]
This looks good to me. Please keep off master until `master` contains a release commit after ff68d336ddd963c681431fbd6d1d014526593842. I’ll lift my “request changes” block at that time.

File: packages/marshal/src/marshal.js [hunk 1]
```diff
@@ -484,6 +484,7 @@ export function makeMarshal(
     const rawTree = harden(JSON.parse(data.body));
     const fullRevive = makeFullRevive(data.slots);
     const result = harden(fullRevive(rawTree));
+    // See https://github.com/Agoric/agoric-sdk/issues/4302
```
Comment (mhofman) [inline packages/marshal/src/marshal.js:487]: I'm actually still writing a more general issue for this.

Comment (mhofman) [reply packages/marshal/src/marshal.js:487]: https://github.com/Agoric/agoric-sdk/issues/4337

Comment (erights) [reply packages/marshal/src/marshal.js]: Done. Linked to https://github.com/Agoric/agoric-sdk/issues/4337 . But this PR only solves the specific #4302 problem.

Comment (mhofman) [reply packages/marshal/src/marshal.js]: How so? Doesn't it prevent all things that produce non passable, like buggy convertSlotToVal ?

**Review by mhofman — APPROVED** [review]

File: packages/marshal/src/marshal.js (hunk 1 above)
Comment (erights) [reply packages/marshal/src/marshal.js]: ***OOPS***. I just merged https://github.com/Agoric/agoric-sdk/pull/4304 which contained changes to two files in the `marshal` package of agoric-sdk. Thus, I made them out of sync.

@kriskowal How should we handle this?

Comment (erights) [reply packages/marshal/src/marshal.js]: I noticed this in preparing to answer @mhofman 's question about #4337

In answer to that question, yes, the assert will catch that specific problem. But mysteriously. I'd like to put more checking directly around the convertSlotToVal call so I can complain with a more specific diagnostic. That same check can ensure that the result is only remotables and promises, which this PR does not enforce.

Comment (mhofman) [reply packages/marshal/src/marshal.js]: Oh yes! Which is why I also wrote https://github.com/Agoric/agoric-sdk/issues/4336 for convertSlotToVal

https://github.com/Agoric/agoric-sdk/issues/4337 is the meta issue listing all known occurrences of non passables that this PR prevents.

**Review by kriskowal — APPROVED** [review]
`master` is released and clear for changes.

**Pull request merged**